src/settings/login/secondfactor/SecondFactorEditDialog.ts

```diff
@@ -44,17 +44,23 @@ export class SecondFactorEditDialog {
 			child: {
 				view: () => this.render(),
 			},
-			okAction: () => this.model.save((user) => {
-				this.dialog.close()
-				SecondFactorEditDialog.showRecoveryInfoDialog(user)
-			}),
+			okAction: () => showProgressDialog(
+				"pleaseWait_msg",
+				// make sure any error messages displayed by save are shown on top of the progress dialog
+				delay(1).then(() => this.model.save((u) => this.finalize(u)))
```

Comment:
we do we want error dialogs on top of progress dialogs? Don't we want it the other way around?